servicetalk-http-api/src/main/java/io/servicetalk/http/api/Uri3986.java

```diff
@@ -196,6 +200,18 @@ final class Uri3986 implements Uri {
         this.uri = uri;
     }
 
+    private static String getLowerCaseScheme(String scheme) {
+        if (regionMatches(scheme, true, 0, HTTP_SCHEME, 0, HTTP_SCHEME.length())) {
+            if (scheme.length() == 4) {
```

Comment:
To avoid expensive `regionMatches` operation in case of unknown scheme, consider first checking the length. If it's 4, run `regionMatches` for HTTP, and if it's 5 - for HTTPS. Consider using `scheme.regionMatches(...)` instead of `CharSequences`, because we know both are `String`s -> no need to go through an extra `instanceof` check.